src/Symfony/Component/VarDumper/Caster/XmlReaderCaster.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * Casts XmlReader class to array representation.
+ *
+ * @author Baptiste Clavié <clavie.b@gmail.com>
+ */
+class XmlReaderCaster
+{
+    private static $nodeTypes = array(
+        \XmlReader::NONE => 'NONE',
+        \XmlReader::ELEMENT => 'ELEMENT',
+        \XmlReader::ATTRIBUTE => 'ATTRIBUTE',
+        \XmlReader::TEXT => 'TEXT',
+        \XmlReader::CDATA => 'CDATA',
+        \XmlReader::ENTITY_REF => 'ENTITY_REF',
+        \XmlReader::ENTITY => 'ENTITY',
+        \XmlReader::PI => 'PI (Processing Instruction)',
+        \XmlReader::COMMENT => 'COMMENT',
+        \XmlReader::DOC => 'DOC',
+        \XmlReader::DOC_TYPE => 'DOC_TYPE',
+        \XmlReader::DOC_FRAGMENT => 'DOC_FRAGMENT',
+        \XmlReader::NOTATION => 'NOTATION',
+        \XmlReader::WHITESPACE => 'WHITESPACE',
+        \XmlReader::SIGNIFICANT_WHITESPACE => 'SIGNIFICANT_WHITESPACE',
+        \XmlReader::END_ELEMENT => 'END_ELEMENT',
+        \XmlReader::END_ENTITY => 'END_ENTITY',
+        \XmlReader::XML_DECLARATION => 'XML_DECLARATION',
+    );
+
+    public static function castXmlReader(\XmlReader $reader, array $a, Stub $stub, $isNested)
+    {
+        $props = Caster::PREFIX_VIRTUAL.'parserProperties';
+        $info = array(
+            'localName' => $reader->localName,
+            'prefix' => $reader->prefix,
+            'nodeType' => new ConstStub(self::$nodeTypes[$reader->nodeType], $reader->nodeType),
+            'depth' => $reader->depth,
+            'isDefault' => $reader->isDefault,
+            'isEmptyElement' => \XmlReader::NONE === $reader->nodeType ? null : $reader->isEmptyElement,
+            'xmlLang' => $reader->xmlLang,
+            'attributeCount' => $reader->attributeCount,
```

Comment:
attributes are not dumped anymore (because we can't get their name, and also because arguments are not directly part of the state of the current iteration)